federation/plugin/cmd/federated-scheduler/app/options/options.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package options provides the federated-scheduler flags
+package options
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	"k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1"
+	"k8s.io/kubernetes/pkg/client/leaderelection"
+	"k8s.io/kubernetes/federation/plugin/pkg/federated-scheduler/factory"
+
+	"github.com/spf13/pflag"
+)
+
+// SchedulerServer has all the context and params needed to run a Scheduler
+type SchedulerServer struct {
```

Comment:
This seems to be a direct copy of https://github.com/kubernetes/kubernetes/blob/master/plugin/cmd/kube-scheduler/app/options/options.go.

We should reuse code rather than copy.
We can move this code to a common place and use it from both kube-scheduler and federation-scheduler.
For ex: we have pkg/genericapiserver for common code between kube-apiserver and federated-apiserver.
Maybe move the common scheduler code to plugin/scheduler?

cc @davidopp